rllib/evaluation/sample_collector.py

```diff
@@ -0,0 +1,172 @@
+from abc import abstractmethod, ABCMeta
+import logging
+from typing import Dict, Optional
+
+from ray.rllib.evaluation.episode import MultiAgentEpisode
+from ray.rllib.models.modelv2 import ModelV2
+from ray.rllib.utils.types import AgentID, EpisodeID, EnvID, PolicyID, \
+    TensorType
+
+logger = logging.getLogger(__name__)
+
+
+class _SampleCollector(metaclass=ABCMeta):
+    """Collects samples for all policies and agents from a multi-agent env.
+
+    Note: This is an experimental class only used when
+    `config._use_trajectory_view_api` = True.
+    Once `_use_trajectory_view_api` becomes the default in configs:
+    This class will deprecate the `SampleBatchBuilder` and
+    `MultiAgentBatchBuilder` classes.
+
+    This API is controlled by RolloutWorker objects to store all data
+    generated by Environments and Policies/Models during rollout and
+    postprocessing. Also provides a method to generate input_dicts for
+    action calculation and train batches from the collected data.
```

Comment:
The comment could be improved to better motivate the reason for this method (to provide performance benefits).

Reply:
done